metrics/web/test/base/presentation/widgets/decorated_container_test.dart

```diff
@@ -0,0 +1,221 @@
+
+        await tester.pumpWidget(
+          const _DecoratedContainerTestbed(margin: margin),
+        );
+
+        final container = tester.widget<Container>(containerFinder);
+
+        expect(container.margin, equals(margin));
+      },
+    );
+
+    testWidgets(
+      "delegates the given padding to the container",
+      (tester) async {
+        const padding = EdgeInsets.all(10.0);
+
+        await tester.pumpWidget(
+          const _DecoratedContainerTestbed(padding: padding),
+        );
+
+        final container = tester.widget<Container>(containerFinder);
+
+        expect(container.padding, equals(padding));
+      },
+    );
+
+    testWidgets(
+      "delegates the given width to the container",
+      (tester) async {
+        const width = 10.0;
+
+        await tester.pumpWidget(
+          const _DecoratedContainerTestbed(width: width),
+        );
+
+        final container = tester.widget<Container>(containerFinder);
+
+        expect(container.constraints.maxWidth, equals(width));
+        expect(container.constraints.minWidth, equals(width));
+      },
+    );
+
+    testWidgets(
+      "delegates the given height to the container",
+      (tester) async {
+        const height = 10.0;
+
+        await tester.pumpWidget(
+          const _DecoratedContainerTestbed(height: height),
+        );
+
+        final container = tester.widget<Container>(containerFinder);
+
+        expect(container.constraints.maxHeight, equals(height));
+        expect(container.constraints.minHeight, equals(height));
+      },
+    );
+
+    testWidgets(
+      "delegates the given constraints to the container",
+      (tester) async {
+        const constraints = BoxConstraints(maxHeight: 10.0);
+
+        await tester.pumpWidget(
+          const _DecoratedContainerTestbed(constraints: constraints),
+        );
+
+        final container = tester.widget<Container>(containerFinder);
+
+        expect(container.constraints, equals(constraints));
+      },
+    );
+
+    testWidgets(
+      "does not add a border width to the height and width of its child",
+      (tester) async {
+        const width = 120.0;
+        const height = 100.0;
+        const expectedSize = Size(width, height);
+        const sizedBox = SizedBox.expand();
+
+        await tester.pumpWidget(
+          _DecoratedContainerTestbed(
+            decoration: BoxDecoration(border: Border.all(width: 10.0)),
+            width: width,
+            height: height,
+            child: sizedBox,
+          ),
+        );
+
+        final actualSize = tester.getSize(find.byType(SizedBox));
+
+        expect(actualSize, equals(expectedSize));
+      },
+    );
+  });
+}
+
+/// A testbed class required to test the [DecoratedContainer] widget.
+class _DecoratedContainerTestbed extends StatelessWidget {
+  /// A default child widget used in tests.
+  static const Widget _defaultChild = Text('default text');
+
+  /// A [Decoration] of this container.
```

Comment:
```suggestion
  /// A [Decoration] of the [DecoratedContainer].
```

Reply:
What do you think about it?

Reply:
The [effective dart](https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance) style guides advice to use `this` instead of the” to refer to a member’s instance, so I think it will be better to leave it as is.

Reply:
In my opinion, as we are already in the DecoratedContainer it will be more suitable to use "this container" because we do not need to navigate to another location by clicking on the widget name in brackets.

Reply:
But we're using `this` inside the testbed, is it okay? Could you please check the colored_bar_test.dart?

Reply:
Maybe we can discuss it?